<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1022</width>
    <height>709</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>RayTracer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <layout class="QGridLayout" name="mainLayout" rowstretch="1,0,0" columnstretch="1,0">
      <item row="1" column="1">
       <widget class="QTabWidget" name="tabWidget_props">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <property name="documentMode">
         <bool>false</bool>
        </property>
        <widget class="QWidget" name="tab_general">
         <attribute name="title">
          <string>General</string>
         </attribute>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QScrollArea" name="scrollArea">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Ignored">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="autoFillBackground">
             <bool>false</bool>
            </property>
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Plain</enum>
            </property>
            <property name="widgetResizable">
             <bool>true</bool>
            </property>
            <widget class="QWidget" name="scrollAreaWidgetContents">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>-211</y>
               <width>365</width>
               <height>404</height>
              </rect>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="autoFillBackground">
              <bool>true</bool>
             </property>
             <property name="styleSheet">
              <string notr="true"/>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_4">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QGroupBox" name="groupBox_2">
                <property name="title">
                 <string>Active Projector</string>
                </property>
                <layout class="QFormLayout" name="formLayout_6">
                 <item row="0" column="1">
                  <widget class="QComboBox" name="comboBox_active_projector"/>
                 </item>
                 <item row="0" column="0">
                  <widget class="QLabel" name="label_13">
                   <property name="text">
                    <string>Object</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QGroupBox" name="groupBox_3">
                <property name="title">
                 <string>Ambient Lighting</string>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_6">
                 <item>
                  <widget class="QCheckBox" name="checkBox_ambient_lighting">
                   <property name="text">
                    <string>Enable</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QFrame" name="frame_lighting">
                   <property name="frameShape">
                    <enum>QFrame::StyledPanel</enum>
                   </property>
                   <property name="frameShadow">
                    <enum>QFrame::Plain</enum>
                   </property>
                   <layout class="QFormLayout" name="formLayout_7">
                    <item row="0" column="0">
                     <widget class="QLabel" name="label_14">
                      <property name="text">
                       <string>Emission</string>
                      </property>
                     </widget>
                    </item>
                    <item row="0" column="1">
                     <widget class="QDoubleSpinBox" name="doubleSpinBox_ambient_lighting"/>
                    </item>
                    <item row="1" column="0">
                     <widget class="QPushButton" name="pushButton_ambient_lighting_color">
                      <property name="text">
                       <string>Color</string>
                      </property>
                     </widget>
                    </item>
                    <item row="1" column="1">
                     <widget class="QLabel" name="label_ambient_lighting_color">
                      <property name="styleSheet">
                       <string notr="true">background-color: rgb(255, 255, 255);</string>
                      </property>
                      <property name="frameShape">
                       <enum>QFrame::Box</enum>
                      </property>
                      <property name="text">
                       <string/>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QGroupBox" name="groupBox">
                <property name="title">
                 <string>Environment Map</string>
                </property>
                <property name="flat">
                 <bool>false</bool>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_5">
                 <item>
                  <widget class="QCheckBox" name="checkBox_env_map">
                   <property name="text">
                    <string>Enable</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QFrame" name="frame_env_map">
                   <property name="frameShape">
                    <enum>QFrame::StyledPanel</enum>
                   </property>
                   <property name="frameShadow">
                    <enum>QFrame::Plain</enum>
                   </property>
                   <layout class="QFormLayout" name="formLayout_5">
                    <item row="0" column="0">
                     <widget class="QLabel" name="label_11">
                      <property name="text">
                       <string>Size</string>
                      </property>
                     </widget>
                    </item>
                    <item row="1" column="0">
                     <widget class="QLabel" name="label_12">
                      <property name="text">
                       <string>Path</string>
                      </property>
                     </widget>
                    </item>
                    <item row="1" column="1">
                     <widget class="QFrame" name="frame">
                      <property name="frameShape">
                       <enum>QFrame::StyledPanel</enum>
                      </property>
                      <property name="frameShadow">
                       <enum>QFrame::Plain</enum>
                      </property>
                      <layout class="QHBoxLayout" name="horizontalLayout">
                       <property name="leftMargin">
                        <number>0</number>
                       </property>
                       <property name="topMargin">
                        <number>0</number>
                       </property>
                       <property name="rightMargin">
                        <number>0</number>
                       </property>
                       <property name="bottomMargin">
                        <number>0</number>
                       </property>
                       <item>
                        <widget class="QLineEdit" name="lineEdit_env_map"/>
                       </item>
                       <item>
                        <widget class="QToolButton" name="toolButton_env_map">
                         <property name="text">
                          <string>...</string>
                         </property>
                        </widget>
                       </item>
                      </layout>
                     </widget>
                    </item>
                    <item row="0" column="1">
                     <widget class="QDoubleSpinBox" name="doubleSpinBox_env_map_size">
                      <property name="decimals">
                       <number>4</number>
                      </property>
                      <property name="minimum">
                       <double>0.001000000000000</double>
                      </property>
                      <property name="maximum">
                       <double>100.000000000000000</double>
                      </property>
                      <property name="singleStep">
                       <double>0.250000000000000</double>
                      </property>
                      <property name="value">
                       <double>10.000000000000000</double>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_transform">
         <attribute name="title">
          <string>Transform</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="3">
           <widget class="QGroupBox" name="groupBox_rotate">
            <property name="title">
             <string>Rotate</string>
            </property>
            <layout class="QFormLayout" name="formLayout_3">
             <item row="2" column="1">
              <widget class="QDoubleSpinBox" name="doubleSpinBox_rotate_y">
               <property name="minimum">
                <double>-360.000000000000000</double>
               </property>
               <property name="maximum">
                <double>360.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.250000000000000</double>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label_7">
               <property name="text">
                <string>x</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label_8">
               <property name="text">
                <string>y</string>
               </property>
              </widget>
             </item>
             <item row="4" column="0">
              <widget class="QLabel" name="label_9">
               <property name="text">
                <string>z</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QDoubleSpinBox" name="doubleSpinBox_rotate_x">
               <property name="minimum">
                <double>-360.000000000000000</double>
               </property>
               <property name="maximum">
                <double>360.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.250000000000000</double>
               </property>
              </widget>
             </item>
             <item row="4" column="1">
              <widget class="QDoubleSpinBox" name="doubleSpinBox_rotate_z">
               <property name="minimum">
                <double>-360.000000000000000</double>
               </property>
               <property name="maximum">
                <double>360.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.250000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QGroupBox" name="groupBox_move">
            <property name="title">
             <string>Move</string>
            </property>
            <layout class="QFormLayout" name="formLayout">
             <item row="2" column="1">
              <widget class="QDoubleSpinBox" name="doubleSpinBox_move_y">
               <property name="minimum">
                <double>-100.000000000000000</double>
               </property>
               <property name="maximum">
                <double>100.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.250000000000000</double>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label">
               <property name="text">
                <string>x</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>y</string>
               </property>
              </widget>
             </item>
             <item row="4" column="0">
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>z</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QDoubleSpinBox" name="doubleSpinBox_move_x">
               <property name="minimum">
                <double>-100.000000000000000</double>
               </property>
               <property name="maximum">
                <double>100.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.250000000000000</double>
               </property>
              </widget>
             </item>
             <item row="4" column="1">
              <widget class="QDoubleSpinBox" name="doubleSpinBox_move_z">
               <property name="minimum">
                <double>-100.000000000000000</double>
               </property>
               <property name="maximum">
                <double>100.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.250000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QGroupBox" name="groupBox_scale">
            <property name="title">
             <string>Scale</string>
            </property>
            <layout class="QFormLayout" name="formLayout_2">
             <item row="2" column="1">
              <widget class="QDoubleSpinBox" name="doubleSpinBox_scale_y">
               <property name="minimum">
                <double>-100.000000000000000</double>
               </property>
               <property name="maximum">
                <double>100.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.250000000000000</double>
               </property>
               <property name="value">
                <double>1.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>x</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label_5">
               <property name="text">
                <string>y</string>
               </property>
              </widget>
             </item>
             <item row="4" column="0">
              <widget class="QLabel" name="label_6">
               <property name="text">
                <string>z</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QDoubleSpinBox" name="doubleSpinBox_scale_x">
               <property name="minimum">
                <double>-100.000000000000000</double>
               </property>
               <property name="maximum">
                <double>100.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.250000000000000</double>
               </property>
               <property name="value">
                <double>1.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="4" column="1">
              <widget class="QDoubleSpinBox" name="doubleSpinBox_scale_z">
               <property name="minimum">
                <double>-100.000000000000000</double>
               </property>
               <property name="maximum">
                <double>100.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.250000000000000</double>
               </property>
               <property name="value">
                <double>1.000000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="1" column="0" colspan="4">
           <widget class="QPushButton" name="pushButton_transform_apply">
            <property name="text">
             <string>Apply</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_emission">
         <attribute name="title">
          <string>Emission</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QCheckBox" name="checkBox_emission">
            <property name="text">
             <string>Enable</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame_emission">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Plain</enum>
            </property>
            <layout class="QFormLayout" name="formLayout_4">
             <item row="0" column="0">
              <widget class="QLabel" name="label_10">
               <property name="text">
                <string>Emission</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QDoubleSpinBox" name="doubleSpinBox_emission"/>
             </item>
             <item row="1" column="0">
              <widget class="QPushButton" name="pushButton_emission_color">
               <property name="text">
                <string>Color</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLabel" name="label_emission_color">
               <property name="styleSheet">
                <string notr="true">background-color: rgb(255, 255, 255);</string>
               </property>
               <property name="frameShape">
                <enum>QFrame::Box</enum>
               </property>
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QGroupBox" name="groupBox_object">
        <property name="title">
         <string>Objects</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QListWidget" name="listWidget_object"/>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_object_delete">
           <property name="text">
            <string>Delete</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QPushButton" name="pushButton_viewport">
        <property name="text">
         <string>Preview</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" rowspan="2">
       <widget class="QGroupBox" name="groupBox_material">
        <property name="title">
         <string>Materials</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="2">
          <widget class="QPushButton" name="pushButton_material_add">
           <property name="text">
            <string>Add</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QPushButton" name="pushButton_material_delete">
           <property name="text">
            <string>Delete</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1" rowspan="3">
          <widget class="QListWidget" name="listWidget_material">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Sunken</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QFrame" name="frame_status">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Sunken</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QLabel" name="label_status">
         <property name="font">
          <font>
           <pointsize>9</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Status:</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1022</width>
     <height>28</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuRender">
    <property name="title">
     <string>Render</string>
    </property>
    <addaction name="actionPreferences"/>
    <addaction name="separator"/>
    <addaction name="actionRun"/>
    <addaction name="actionRun_in_viewport"/>
   </widget>
   <widget class="QMenu" name="menuAdd">
    <property name="title">
     <string>Add</string>
    </property>
    <widget class="QMenu" name="menuPolygon_Model">
     <property name="title">
      <string>Polygon Model</string>
     </property>
     <addaction name="actionLoad_2"/>
    </widget>
    <addaction name="separator"/>
    <addaction name="actionSphere"/>
    <addaction name="actionCube"/>
    <addaction name="actionCilinder"/>
    <addaction name="actionCone"/>
    <addaction name="actionPlne"/>
    <addaction name="actionDisl"/>
    <addaction name="actionNull_Object"/>
    <addaction name="separator"/>
    <addaction name="menuPolygon_Model"/>
    <addaction name="separator"/>
    <addaction name="actionCamera"/>
   </widget>
   <addaction name="menuAdd"/>
   <addaction name="menuRender"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionSphere">
   <property name="text">
    <string>Sphere</string>
   </property>
  </action>
  <action name="actionCube">
   <property name="text">
    <string>Cube</string>
   </property>
  </action>
  <action name="actionPlne">
   <property name="text">
    <string>Plane</string>
   </property>
  </action>
  <action name="actionCone">
   <property name="text">
    <string>Cone</string>
   </property>
  </action>
  <action name="actionCilinder">
   <property name="text">
    <string>Cylinder</string>
   </property>
  </action>
  <action name="actionDisl">
   <property name="text">
    <string>Disk</string>
   </property>
  </action>
  <action name="actionLoad_2">
   <property name="text">
    <string>Load</string>
   </property>
  </action>
  <action name="actionPreferences">
   <property name="text">
    <string>Preferences</string>
   </property>
  </action>
  <action name="actionRun">
   <property name="text">
    <string>Run</string>
   </property>
  </action>
  <action name="actionRun_in_viewport">
   <property name="text">
    <string>Run in viewport</string>
   </property>
  </action>
  <action name="actionCamera">
   <property name="text">
    <string>Camera</string>
   </property>
  </action>
  <action name="actionNull_Object">
   <property name="text">
    <string>Null Object</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
